tests/rules/filterrequestfromurl_test.py

```diff
@@ -0,0 +1,167 @@
+# as published by the Free Software Foundation; either version 2
+# of the License.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program; if not, write to the Free Software
+# Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+
+
+from unittest import TestCase
+
+from hammertime.rules import FilterRequestFromURL
+from hammertime.http import Entry
+from hammertime.ruleset import RejectRequest
+from tests.fixtures import async_test
+
+
+class TestFilterRequestFromURL(TestCase):
+
+    def setUp(self):
+        pass
+
+    @async_test()
+    async def test_before_request_allow_request_to_host_matching_filter_in_whitelist(self):
+        filter = FilterRequestFromURL(allowed_urls="example.com")
+
+        try:
+            await filter.before_request(Entry.create("http://example.com/"))
+        except RejectRequest as e:
+            self.fail(str(e))
+
+    @async_test()
+    async def test_before_request_reject_request_to_host_not_matching_filter_in_whitelist(self):
+        filter = FilterRequestFromURL(allowed_urls="example.com")
+
+        with self.assertRaises(RejectRequest):
+            await filter.before_request(Entry.create("http://www.example.com/"))
+
+    @async_test()
+    async def test_before_request_accept_list_of_url_for_whitelist(self):
+        filter = FilterRequestFromURL(allowed_urls=["example.com", "www.example.com", "test.example.com"])
+        good_urls = ["http://example.com/", "https://www.example.com/index.php", "http://test.example.com/index.html"]
+        bad_urls = ["http://abc.example.com/", "https://example.ca/", "http://www.example.com.net/"]
+
+        for url in bad_urls:
+            with self.assertRaises(RejectRequest):
+                await filter.before_request(Entry.create(url))
+        for url in good_urls:
+            try:
+                await filter.before_request(Entry.create(url))
+            except RejectRequest as e:
+                self.fail(str(e))
+
+    @async_test()
+    async def test_before_request_allow_request_to_path_matching_filter_in_whitelist(self):
+        filter = FilterRequestFromURL(allowed_urls="/allowed")
+
+        try:
+            await filter.before_request(Entry.create("http://example.com/allowed/index.html"))
+        except RejectRequest as e:
+            self.fail(str(e))
+
+    @async_test()
+    async def test_before_request_reject_request_to_path_not_matching_filter_in_whitelist(self):
+        filter = FilterRequestFromURL(allowed_urls="/allowed")
+
+        with self.assertRaises(RejectRequest):
+            await filter.before_request(Entry.create("http://example.com/test/index.html"))
```

Comment:
What about http://example.com/allowed-oops ?

Reply:
Now fixed and tested by https://github.com/delvelabs/hammertime/pull/25/files#diff-bd6cbceded6380d894ad8478634e1dccR138